### DISCUSSION TITLE
Show loader while async function fetches data

### DISCUSSION THREAD

**Discussion by talkoren** [jspsych/jsPsych#3234] [Q&A] [answered]
Hello,

My experiment includes a trial which uses the `jsPsychCallFunction` plugin with `async: true` to fetch data from a server (it does so successfully). Is there a way to present a loader gif while the function waits for a response?

Thank you!

**Comment by Shaobin-Jiang** [comment, marked as answer]
The example uses a timeout and a line of text instead of a request to the server and the loader gif, but things should work the same way.

```javascript
let jsPsych = initJsPsych();

let timeline = [];

let trial = {
    type: jsPsychCallFunction,
    func: function (done) {
        setTimeout(function () {
            jsPsych.getDisplayElement().innerHTML = '';
            done();
        }, 5000);
    },
    async: true,
    on_start: function () {
        jsPsych.getDisplayElement().innerHTML = 'Please wait for 5 seconds...';
    },
};

jsPsych.run([trial]);
```

**Reply by talkoren** [reply]
Works perfectly, thank you @Shaobin-Jiang!
